assisted_deployment.sh

```diff
@@ -273,26 +281,39 @@ function patch_extra_host_manifests() {
 }
 
 function install_assisted_service() {
- mkdir -p "${ASSETS_DIR}"
- patch_extra_host_manifests
- deploy_local_storage
- deploy_hive
- deploy_assisted_operator
+  install_prerequisites_assisted_service
+  deploy_assisted_operator
 
- oc wait -n "$ASSISTED_NAMESPACE" --for=condition=Ready pod -l app=assisted-service --timeout=90s
+  oc wait -n "$ASSISTED_NAMESPACE" --for=condition=Ready pod -l app=assisted-service --timeout=90s
+
+  echo "Installation finished..."
+  echo "For debugging purposes all the manifests have been saved in ${ASSETS_DIR}"
+  echo "Please remember to manually apply BareMetalHost manifest available in the directory above. You can use the following command:"
+  echo "oc apply -f ${ASSETS_DIR}/06-extra-host-manifests.yaml"
+}
+
+# For a development workflow where we want to deploy the assisted service using operator-sdk it is
+# useful to have a process installing required dependencies, i.e. LSO and Hive as well as generating
+# AgentServiceConfig manifest that is required later on.
+function install_prerequisites_assisted_service() {
+  mkdir -p "${ASSETS_DIR}"
+  patch_extra_host_manifests
+  deploy_local_storage
+  deploy_hive
+  generate_assisted_operator
```

Comment:
Does the operator-sdk deployment use all of the artifacts from generate_assisted_operator? If it only uses the AgentServiceConfig, perhaps that should be separated from generate_assisted_operator? Or the operator specific parts of generate_assisted_operator should be moved to deploy_assisted_operator?

Reply:
Ok, I split `generate_assisted_service_config` out of the `generate_assisted_operator` so that in operator-sdk prerequisites we generate only AgentServiceConfig. Other resources like CatalogSource, OperatorGroup and Subscription are not needed to be created as a manifest because operator-sdk takes care of them, so indeed AgentServiceConfig is the only resource we need in the manual flow.